src/promptflow/promptflow/executor/_line_execution_process_pool.py

```diff
@@ -330,6 +333,21 @@ def _monitor_workers_and_process_tasks_in_thread(
 
             self._processing_idx.pop(line_number)
 
+        # End the process when the batch timeout is exceeded.
+        self._processes_manager.end_process(index)
```

Comment:
We have already stopped them in the loop?

Reply:
Updated.